internal/pkg/file/file.go

```diff
@@ -33,6 +36,49 @@ func Exists(fileName string) bool {
 	return !info.IsDir()
 }
 
+type DottedBaseName int
+
+const (
+	// NotDotted indicates that a path is not considered to be hidden on
+	// Unix-like operating systems, i.e. it doesn't start with a dot.
+	NotDotted DottedBaseName = iota + 1
+
+	// Dotted indicates that a path is considered to be hidden on Unix-like
+	// operating systems, i.e. it starts with a dot.
+	Dotted
+
+	// RelativeBase indicates that it cannot be decided if the path is
+	// considered to be hidden on Unix-like operating systems or not, since it's
+	// base name is relative.
+	RelativeBase
+)
+
+func (d DottedBaseName) String() string {
+	switch d {
+	case NotDotted:
+		return "NotDotted"
+	case Dotted:
+		return "Dotted"
+	case RelativeBase:
+		return "RelativeBase"
+	}
+
+	return strconv.FormatInt(int64(d), 10)
+}
+
+// IsDottedBaseName checks if the given path is considered to be hidden on
+// Unix-like operating systems, i.e. if its base name starts with a dot.
+func IsDottedBaseName(path string) DottedBaseName {
```

Comment:
The naming feels bit off. IMO the logic is more like converting a string path to `DottedBaseName` type so in my head something like `ToDottedBase` feels more natural.

Reply:
I see your point. Not too fond of the name (and the API) either. I previously called this `IsHiddenPath`, but that was also not feeling right. The initial implementation used a boolean return type, until I realized during implementation that a tri-state return value would be better. I refrained from using `(bool, bool)` which also felt awkward, somehow.

I don't think that the `To` prefix would be a better fit, since it's not really a conversion, too. What about `Get` though? Maybe `GetDotPrefixStyle` or something in that direction?

Reply:
> Maybe GetDotPrefixStyle or something in that direction?

sounds much better.